### PR TITLE
Move some tests to more reasonable directories - 9.5

### PR DESCRIPTION
**Pull request by c410-f3r**
cc #73494
r? @petrochenkov

### CODE REVIEW

**Comment by petrochenkov** [comment]
@c410-f3r 
Could you submit the parser -> lexer move as a separate PR?
It's sufficiently different from the remaining changes, and it's hard to review PRs with hundreds of moved tests anyway.

**Comment by c410-f3r** [comment]
> @c410-f3r Could you submit the parser -> lexer move as a separate PR? It's sufficiently different from the remaining changes, and it's hard to review PRs with hundreds of moved tests anyway.

Done

File: src/tools/tidy/src/ui_tests.rs [hunk 1]
```diff
@@ -7,9 +7,8 @@ use std::path::Path;
 
 const ENTRY_LIMIT: usize = 1000;
 // FIXME: The following limits should be reduced eventually.
-const ROOT_ENTRY_LIMIT: usize = 1102;
```
Comment (petrochenkov) [inline src/tools/tidy/src/ui_tests.rs:10]: The goal for the root and `issues` is to be empty, so we need to keep the restrictions, even if they are less than 1000.

**Comment by petrochenkov** [comment]
with-no-generics-in-parameter-list.rs -> parser
no-mangle-associated-fn.rs -> extern (or foreign, the subdirectories duplicate each other)
byte-literals.rs -> array-slice-vec
ignore-all-the-things.rs -> pattern
placement-syntax.rs -> parser
tail-direct.rs -> can be removed now
blind-item-mixed-crate-use-item.rs -> resolve
blind-item-mixed-use-item.rs -> resolve
raw-fat-ptr.rs -> stdlib-unit-tests
link-cfg-works.rs -> linkage-attr (for now)
impl-privacy-xc-2.rs -> privacy
reachable-unnameable-items.rs -> privacy
leak-unique-as-tydesc.rs -> can be removed now
pure-sum.rs -> can be removed now 
volatile-fat-ptr.rs -> stdlib-unit-tests
unify-return-ty.rs -> typeck
no-std-1.rs -> resolve
no-std-2.rs -> resolve
no-std-3.rs -> resolve
thread-local-not-in-prelude.rs -> hygiene
llvm-pr32379.rs -> codegen maybe
backtrace-debuginfo.rs -> runtime
matches2021.rs -> stdlib-unit-tests
atomic-print.rs -> runtime or threads-sendsync
init-large-type.rs -> codegen
clone-with-exterior.rs -> threads-sendsync
dyn-trait-compatibility.rs -> parser
export-tag-variant.rs -> privacy
export.rs -> can be removed now
pathless-extern-ok.rs -> resolve
float-literal-inference-restrictions.rs -> mismatched_types
kindck-implicit-close-over-mut-var.rs -> borrowck or closures
minmax-stability-issue-23687.rs -> stdlib-unit-tests
retslot-cast.rs -> coercion
syntax-trait-polarity-feature-gate.rs -> can be removed now
ufcs-type-params.rs -> typeck
not-sync.rs -> stdlib-unit-tests
synthetic-param.rs -> impl-trait


The `blind` directory doesn't make sense and its content should be moved to `resolve`.
I'm again convinced that the `binding` directory doesn't make sense and needs to be dismantled.

**Comment by c410-f3r** [comment]
Should be good now

**Comment by petrochenkov** [comment]
impl-privacy-xc-2.rs -> privacy

@bors r+

**Pull request merged**